### DISCUSSION TITLE
Activation of DEBUG log

### DISCUSSION THREAD

**Discussion by flefevre** [open-webui/open-webui#1335] [Q&A] [answered]
Hello 
is there any OPENWEBUI environemnt variabel to generate more log? 
I have some difficulties to start it, i will appreciate to have details log.
Thanks
François

**Comment by ghost** [comment, marked as answer]
Hi!

There is an environment variable available to enable elevated logging. I've submitted a pull request to update the official documentation here: open-webui/docs#31

> The default global log level of `INFO` can be overridden with the `GLOBAL_LOG_LEVEL` environment variable. When set, this executes a [basicConfig](https://docs.python.org/3/library/logging.html#logging.basicConfig) statement with the `force` argument set to *True* within `config.py`. This results in reconfiguration of all attached loggers:
>> _If this keyword argument is specified as true, any existing handlers attached to the root logger are removed and closed, before carrying out the configuration as specified by the other arguments._
>
> In addition to all Open-WebUI `log()` statements, this also affects any imported Python modules that use the Python Logging module `basicConfig` mechanism including [urllib](https://docs.python.org/3/library/urllib.html).
>
> For example, set `DEBUG` logging level as a Docker parameter:
> ```
>--env GLOBAL_LOG_LEVEL="DEBUG"
> ```

**Reply by flefevre** [reply]
Thank you. It works.

**Comment by therealmichaelberna** [comment]
Additional note for anyone trying to locate the logs:

If this has been installed via docker, you can view the logs of open webui or pipelines as follows (adjust name if you have renamed your docker image):
`sudo docker logs open-webui`
`sudo docker logs pipelines`

**Reply by rgaricano** [reply]
& for reference to:

https://docs.openwebui.com/getting-started/advanced-topics/logging
```
Level        When it’s used

DEBUG       Detailed information, typically of interest only when diagnosing problems.
INFO        Confirmation that things are working as expected.
WARNING     An indication that something unexpected happened, or indicative of some problem in the near future (e.g. ‘disk space low’). The software is still working as expected.
ERROR       Due to a more serious problem, the software has not been able to perform some function.
CRITICAL    A serious error, indicating that the program itself may be unable to continue running.
NOTSET      All messages to be processed when the logger is the root logger, or delegation to the parent when the logger is a non-root logger. (Root logger is created with level [WARNING])
```

**Comment by flefevre** [comment]
Thanks
